lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java

```diff
@@ -196,6 +191,58 @@ public void attach_with_channel_params_set_options() {
         }
     }
 
+    @Test
+    public void attached_state_is_correctly_achieved() {
+        String channelName = "test_" + testParams.name;
+        AblyRealtime ably = null;
+        AblyRest ablyRest;
+
+        try {
+            ClientOptions opts = createOptions(testVars.keys[0].keyStr);
+            ably = new AblyRealtime(opts);
+            ablyRest = new AblyRest(opts);
+
+            /* wait until connected */
+            (new ConnectionWaiter(ably.connection)).waitFor(ConnectionState.connected);
+            assertEquals("Verify connected state reached", ably.connection.state, ConnectionState.connected);
+
+            final Map<String, String> params = new HashMap<>();
+            params.put("rewind", "1");
+            final ChannelOptions options = new ChannelOptions();
+            options.params = params;
+
+            Channel channel = ably.channels.get(channelName, options);
+            ablyRest.channels.get(channelName).publish("", "test-message");
+
+            // channel is in initialized state when created
+            assertEquals(ChannelState.initialized, channel.state);
+
+            Helpers.MessageWaiter messageWaiter = new Helpers.MessageWaiter(channel);
+            // channel is not attached yet
+            assertEquals(ChannelState.attaching, channel.state);
+
+            messageWaiter.waitFor(1);
+            // when message is received channel state is already in attached state
+            assertEquals(ChannelState.attached, channel.state);
+
+            ably.channels.release(channelName);
+
+            // verify there are no channels after realising
+            assertFalse(ably.channels.entrySet().iterator().hasNext());
+            Thread.sleep(5000);
+
+            // get channel again
+            channel = ably.channels.get(channelName, options);
+            assertEquals(ChannelState.initialized, channel.state);
+        } catch (AblyException | InterruptedException e) {
+            e.printStackTrace();
+            fail("init0: Unexpected exception instantiating library");
```

Comment:
Again, I realise this is a cut+paste of the same error elsewhere, but this message should refer to the correct test name, or not refer to the name at all

Reply:
Fail message reflects on scenario when Ably cannot be constructed. Or are you referring to something else?

Reply:
My comment was just that `init0` was the name of a test. This line of code got copied into multiple other tests, without updating the test name. Either update it, or remove it.